static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js

```diff
@@ -1,159 +1,160 @@
 define([
-    'common/utils/fastdom-promise',
-    'common/utils/$',
+    'qwery',
+    'fastdom',
     'common/modules/navigation/edition-picker',
     'common/modules/navigation/editionalise-menu'
 ], function (
-    fastdomPromise,
-    $,
+    qwery,
+    fastdom,
     editionPicker,
     editionaliseMenu
 ) {
-    var mainMenuId = '#main-menu';
-    var html = $('html');
-    var mainMenuEl = $(mainMenuId);
-    var veggieBurgerLink = $('.js-change-link');
-    var primaryItems = $('.js-close-nav-list');
+    var html = qwery('html')[0];
+    var menuItems = qwery('.js-close-nav-list');
+    var buttonClickHandlers = {
+        'main-menu-toggle': veggieBurgerClickHandler,
+        'edition-picker': editionPicker
+    };
 
     function closeAllOtherPrimaryLists(targetItem) {
-        primaryItems.each(function (item) {
-
+        menuItems.forEach(function (item) {
             if (item !== targetItem) {
                 item.removeAttribute('open');
             }
         });
     }
 
-    function animateMenuOpen() {
-        return fastdomPromise.write(function () {
-            mainMenuEl.addClass('off-screen shown');
+    function enhanceToButton() {
+        var checkboxes = qwery('.js-enhance-checkbox');
+        fastdom.read(function () {
+            var buttons = checkboxes.map(function (checkbox) {
+                var button = document.createElement('button');
+                var checkboxId = checkbox.id;
+                var checkboxControls = checkbox.getAttribute('aria-controls');
+                var forEachClass = Array.prototype.forEach.bind(checkbox.classList);
+
+                forEachClass(function (c) {
```

Comment:
TL;DR this doesn't quite work as one might expect!

Actually this won't work because the context gets broken by `bind`. Even if we rebind the correct context using `button.classList.add.bind(button.classList)`, the `forEach` callback passes the loop index and a reference to the checkbox classList array, which get stringified and added to the button classList.